File: Java/app/src/main/java/com/example/topquiz/model/QuestionBank.java
package com.example.topquiz.model;

import java.util.Collections;
import java.util.List;

public class QuestionBank {
    private List<Question> questions;
    private int nextQuestionIdx;

    public QuestionBank(List<Question> questions) {
        this.questions = questions;
        Collections.shuffle(this.questions);
        nextQuestionIdx = 0;
    }

    public Question getQuestion() {
        if (nextQuestionIdx == questions.size()) {
            nextQuestionIdx = 0;
        }
        return questions.get(nextQuestionIdx++);
    }
}

File: Kotlin/app/src/main/java/com/example/topquiz_kotlin/models/QuestionBank.kt
package com.example.topquiz_kotlin.models
import com.example.topquiz_kotlin.questionList

class QuestionBank() {
    private var nextQuestionIdx = 0
    private val questions = mutableListOf(*questionList)

    init {
        questions.shuffle();
    }

    fun getQuestion() : Question {
        if (nextQuestionIdx == questions.size) nextQuestionIdx = 0
        return questions.get(nextQuestionIdx++)
    }
}

